File: nelio-alves/algoritmo-e-logica-de-programacao/java/src/exercicios/MatrizGeral.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class MatrizGeral {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, linha, coluna;
		double somaPositivos;

	    System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();

	    double[][] mat = new double[N][N];

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
	        	System.out.printf("Elemento [%d,%d]: ", i, j);
				mat[i][j] = sc.nextDouble();
	        }
	    }

		System.out.println("\nMATRIZ DIGITADA:");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print(mat[i][j] + " ");
			}
			System.out.println();
		}

		somaPositivos = 0;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (mat[i][j] > 0) {
					somaPositivos = somaPositivos + mat[i][j];
				}
			}
		}

	    System.out.printf("\nSOMA DOS POSITIVOS: %.1f\n\n", somaPositivos);

	    System.out.print("Escolha uma linha: ");
		linha = sc.nextInt();

		System.out.print("LINHA ESCOLHIDA: ");
		for (int i = 0; i < N; i++) {
	    	System.out.printf("%.1f ", mat[linha][i]);
	    }

	    System.out.print("\n\nEscolha uma coluna: ");
		coluna = sc.nextInt();

		System.out.print("COLUNA ESCOLHIDA: ");
		for (int i = 0; i < N; i++) {
	    	System.out.printf("%.1f ", mat[i][coluna]);
	    }

	    System.out.print("\n\nDIAGONAL PRINCIPAL: ");
		for (int i = 0; i < N; i++) {
	    	System.out.printf("%.1f ", mat[i][i]);
	    }

		for (int i = 0; i < N; i++) {
			for (int j=0; j < N; j++) {
	            if (mat[i][j] < 0) {
					mat[i][j] = Math.pow(mat[i][j], 2);
	            }
	        }
	    }

		System.out.println("\n\nMATRIZ ALTERADA:");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
	        	System.out.printf("%.1f ", mat[i][j]);
	        }
			System.out.println();
	    }

		sc.close();
	}
}

File: nelio-alves/algoritmo-e-logica-de-programacao/python/exercicios/matriz_geral.py
import math

N: int; linha: int; coluna: int
somaPositivos: float

N = int(input("Qual a ordem da matriz? "))

mat: [[float]] = [[0 for x in range(N)] for x in range(N)]

for i in range(N):
	for j in range(N):
		mat[i][j] = float(input(f"Elemento [{i},{j}]: "))

print("\nMATRIZ DIGITADA:")
for i in range(N):
    for j in range(N):
        print(f"{mat[i][j]} ", end="")
    print()

somaPositivos = 0
for i in range(N):
	for j in range(N):
		if mat[i][j] > 0:
			somaPositivos = somaPositivos + mat[i][j]

print(f"\nSOMA DOS POSITIVOS: {somaPositivos:.1f}\n")

linha = int(input("Escolha uma linha: "))

print("LINHA ESCOLHIDA: ", end="");
for i in range(N):
	print(f"{mat[linha][i]:.1f} ", end="")

coluna = int(input("\n\nEscolha uma coluna: "))

print("COLUNA ESCOLHIDA: ", end="");
for i in range(N):
	print(f"{mat[i][linha]:.1f} ", end="")

print("\n\nDIAGONAL PRINCIPAL: ", end="");
for i in range(N):
	print(f"{mat[i][i]:.1f} ", end="")

for i in range(N):
	for j in range(N):
		if mat[i][j] < 0:
			mat[i][j] = math.pow(mat[i][j], 2);

print("\n\nMATRIZ ALTERADA:");
for i in range(N):
	for j in range(N):
		print(f"{mat[i][j]:.1f} ", end="")
	print()